### PR TITLE
AKS Workshop Plan

### PR DESCRIPTION
**Pull request by chzbrgr71**
Added markdown doc with AKS workshop outlines

### CODE REVIEW

File: aks-workshop-plan.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+  * Network policy
+  * Advanced Container Networking Services (ACNS)
+  * Private AKS Clusters
+4. Cost optimization
+  * Node Auto-provision (Karpenter)
+  * VPA
+  * AKS Cost Analysis
+  * Spot and Reservced Instances
+5. Basic security topics
+  * Access management
```
Comment (KenKilty) [inline aks-workshop-plan.md:77]: Kubernetes RBAC? Azure RBAC for AKS?

File: aks-workshop-plan.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+8. GitOps and Application Delivery 
+9. Load Testing your application
+
+### AI Apps on AKS
+
+1. GPU Configuration
+2. MLOps
+3. KAITO
+
+### Securing AKS Clusters
```
Comment (KenKilty) [inline aks-workshop-plan.md:93]: Something like this? https://medium.com/@SecurityArchitect/hardening-azure-kubernetes-essential-settings-for-multitenancy-d2493a4511d4

Comment (chzbrgr71) [reply aks-workshop-plan.md:93]: This one is likely going to be created by Steve Griffith from GBB but we can have input. I put an example of the kind of thing we were thinking in the doc

File: aks-workshop-plan.md (hunk 1 above)
Comment (chzbrgr71) [reply aks-workshop-plan.md:77]: We can ask the PM team what our default should be and start there. We can also show both and give them options. I would lean towards k8s RBAC

File: aks-workshop-plan.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+  * Deployment Safeguards
+  * Managing secrets
+  * Workload Identity
+  * Image security
+6. Fleet Management
+7. Istio Service Mesh Add-on
+8. GitOps and Application Delivery 
+9. Load Testing your application
+
+### AI Apps on AKS
```
[the remark this answers is not in the document]
Comment (chzbrgr71) [reply aks-workshop-plan.md:87]: Lots still to think about here. Let's decide what we want it to be. 

It shouldn't just be a KAITO lab.

File: aks-workshop-plan.md [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+Example: https://moaw.dev/workshop/getting-started-with-aks 
+
+1. Getting started
+    * Objectives
+    * Prerequisites
+    * Workshop instructions
+    * Setting up your environment
+2. Deploy your AKS Cluster
+    * Familiarize with AKS Presets in portal
+    * Deploy AKS Cluster (CLI)
```
Comment (pauldotyu) [inline aks-workshop-plan.md:20]: We should just have folks deploy an Automatic cluster from the portal

File: aks-workshop-plan.md [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+    * Connect to AKS 
+3. Container basics (skip this if in cloud shell)
+    * Dockerfile
+    * Create a container
+    * Run container locally
+4. Azure Container Registry 
+    * Deploy Azure Container Registry (ACR)
+    * Familiarize with ACR
```
Comment (pauldotyu) [inline aks-workshop-plan.md:28]:
```suggestion
    * Attache ACR to AKS cluster
```

File: aks-workshop-plan.md [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+    * Familiarize with ACR 
+    * Push container image (from step 3)
+5. Deploy Store App to AKS
+    * Getting familiar with AKS Store app 
+    * Publishing images to ACR
+    * Deployments and Services
+    * Ingress and App Routing Add-on
+6. App Updates and Rollbacks
+    * Deployment rolling updates
+    * AKS Automated Deployments
```
Comment (pauldotyu) [inline aks-workshop-plan.md:37]: I think this should be in the section above. Currently not all services in the aks-store-demo can be deployed via AKS automated deployments. I believe we should deploy either order-service or product-service using AD

File: aks-workshop-plan.md [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+    * Storage classes
+    * Creating a PVC and adapting app to persist data
+9. Switch Databases to Azure PaaS
+    * Replace RabbitMQ with Azure Service Bus
+    * Replace MongoDB for Azure Cosmos DB
+10. Observability
+    * Prometheus and Grafana
+    * Container Insights
+    * Control plane metrics
+    * App Insights
```
Comment (pauldotyu) [inline aks-workshop-plan.md:52]: Might need to leave this one off as none of the apps are currently writing logs via app insights.

File: aks-workshop-plan.md (hunk 1 above)
Comment (pauldotyu) [reply aks-workshop-plan.md:77]: My 2 cents... we should do both to show the integration between k8s and azure rbac via microsoft entra id (i.e., granting perms for sre with access to all namespaces vs perms for dev in a single namespace). But all depends on the subscription/tenant access a learner would have access to .

File: aks-workshop-plan.md [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+  * Node Auto-provision (Karpenter)
+  * VPA
+  * AKS Cost Analysis
+  * Spot and Reservced Instances
+5. Basic security topics
+  * Access management
+  * Deployment Safeguards
+  * Managing secrets
+  * Workload Identity
+  * Image security
```
Comment (pauldotyu) [inline aks-workshop-plan.md:81]: How far deep should we go here? Just implement Image Cleaner or go all the way with image signing and verification too?

File: aks-workshop-plan.md [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+6. Fleet Management
+7. Istio Service Mesh Add-on
+8. GitOps and Application Delivery
```
Comment (pauldotyu) [inline aks-workshop-plan.md:84]: Flux extension or ArgoCD open-source deployment? My vote is Argo... 😉

File: aks-workshop-plan.md [hunk 10]
```diff
@@ -0,0 +1,108 @@
...
+  * Image security
+6. Fleet Management
+7. Istio Service Mesh Add-on
+8. GitOps and Application Delivery 
+9. Load Testing your application
+
+### AI Apps on AKS
+
+1. GPU Configuration
+2. MLOps
```
Comment (pauldotyu) [inline aks-workshop-plan.md:90]: This could be the "big bertha" build out on azure

File: aks-workshop-plan.md [hunk 11]
```diff
@@ -0,0 +1,108 @@
...
+3. KAITO
+
+### Securing AKS Clusters
+Example: https://securekubernetes.com
+
+Focus on attack and defense approaches
+Compliance
+
+### Skillable Lab for reference (KubeCon Paris)
```
Comment (pauldotyu) [inline aks-workshop-plan.md:99]:
```suggestion
### Existing Lab for reference (KubeCon Paris)
```

File: aks-workshop-plan.md (hunk 9 above)
Comment (chzbrgr71) [reply aks-workshop-plan.md:84]: https://x.com/memenetes/status/1817953369824100626

File: aks-workshop-plan.md [hunk 12]
```diff
@@ -0,0 +1,108 @@
...
+    * Deploy AKS Cluster (CLI)
+    * Connect to AKS 
+3. Container basics (skip this if in cloud shell)
+    * Dockerfile
+    * Create a container
+    * Run container locally
+4. Azure Container Registry 
+    * Deploy Azure Container Registry (ACR)
+    * Familiarize with ACR 
+    * Push container image (from step 3)
```
Comment (chzbrgr71) [inline aks-workshop-plan.md:29]: Use `az acr import`

File: aks-workshop-plan.md [hunk 13]
```diff
@@ -0,0 +1,108 @@
...
+    * Workshop instructions
+    * Setting up your environment
+2. Deploy your AKS Cluster
+    * Familiarize with AKS Presets in portal
+    * Deploy AKS Cluster (CLI)
+    * Connect to AKS 
+3. Container basics (skip this if in cloud shell)
+    * Dockerfile
+    * Create a container
+    * Run container locally
```
Comment (pauldotyu) [inline aks-workshop-plan.md:25]: Remove this and point to quickstart guide: https://learn.microsoft.com/azure/aks/tutorial-kubernetes-prepare-app?tabs=azure-cli

File: aks-workshop-plan.md (hunk 6 above)
Comment (chzbrgr71) [reply aks-workshop-plan.md:37]: We will use a manual deploy in this step and add Auto Deploys and CI/CD in an optional step later

File: aks-workshop-plan.md [hunk 14]
```diff
@@ -0,0 +1,108 @@
...
+5. Deploy Store App to AKS
+    * Getting familiar with AKS Store app 
+    * Publishing images to ACR
+    * Deployments and Services
+    * Ingress and App Routing Add-on
+6. App Updates and Rollbacks
+    * Deployment rolling updates
+    * AKS Automated Deployments
+7. Application and Cluster Scaling
+    * Cluster Autoscaler
```
Comment (chzbrgr71) [inline aks-workshop-plan.md:39]: This won't be needed since Automatic uses NAP

File: aks-workshop-plan.md [hunk 15]
```diff
@@ -0,0 +1,108 @@
...
+    * Push container image (from step 3)
+5. Deploy Store App to AKS
+    * Getting familiar with AKS Store app 
+    * Publishing images to ACR
+    * Deployments and Services
+    * Ingress and App Routing Add-on
+6. App Updates and Rollbacks
+    * Deployment rolling updates
+    * AKS Automated Deployments
+7. Application and Cluster Scaling
```
Comment (chzbrgr71) [inline aks-workshop-plan.md:38]: Add basic scaling steps. Manual deployment scaling and HPA, then KEDA

File: aks-workshop-plan.md (hunk 14 above)
Comment (pauldotyu) [inline aks-workshop-plan.md:39]:
```suggestion
    * Pod disruption budgets
```

File: aks-workshop-plan.md [hunk 16]
```diff
@@ -0,0 +1,108 @@
...
+4. Azure Container Registry 
+    * Deploy Azure Container Registry (ACR)
+    * Familiarize with ACR 
+    * Push container image (from step 3)
+5. Deploy Store App to AKS
+    * Getting familiar with AKS Store app 
+    * Publishing images to ACR
+    * Deployments and Services
+    * Ingress and App Routing Add-on
+6. App Updates and Rollbacks
```
Comment (chzbrgr71) [inline aks-workshop-plan.md:35]: Add step on PDB's here

File: aks-workshop-plan.md (hunk 7 above)
Comment (chzbrgr71) [reply aks-workshop-plan.md:52]: Agreed. Remove

File: aks-workshop-plan.md [hunk 17]
```diff
@@ -0,0 +1,108 @@
...
+    * Deployment rolling updates
+    * AKS Automated Deployments
+7. Application and Cluster Scaling
+    * Cluster Autoscaler
+    * Setting request and limits
+    * Scaling with KEDA based on CPU utilization
+8. Storage and Persisting Data
```
Comment (chzbrgr71) [inline aks-workshop-plan.md:42]: Keep 8 and 9 for now. Possibly combine

**Pull request merged**